contrib/node/src/python/pants/contrib/node/subsystems/command.py

```diff
@@ -0,0 +1,92 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import logging
+import os
+from collections import namedtuple
+
+from pants.util.process_handler import subprocess
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(namedtuple('Command', ['executable', 'args', 'extra_paths'])):
+  """Describes a command to be run using a Node distribution."""
+
+  @property
+  def cmd(self):
+    """The command line that will be executed when this command is spawned.
+
+    :returns: The full command line used to spawn this command as a list of strings.
+    :rtype: list
+    """
+    return [self.executable] + (self.args or [])
+
+  def _prepare_env(self, kwargs):
+    """Returns a modifed copy of kwargs['env'], and a copy of kwargs with 'env' removed.
+
+    If there is no 'env' field in the kwargs, os.environ.copy() is used.
+    env['PATH'] is set/modified to contain the Node distribution's bin directory at the front.
+
+    :param kwargs: The original kwargs.
+    :returns: An (env, kwargs) tuple containing the modified env and kwargs copies.
+    :rtype: (dict, dict)
+    """
+    kwargs = kwargs.copy()
+    env = kwargs.pop('env', os.environ).copy()
+    env['PATH'] = os.path.pathsep.join(self.extra_paths + [env.get('PATH', '')])
```

Comment:
I added a method [`get_joined_path()` to contextutil.py](https://github.com/pantsbuild/pants/blob/8212fd1615c1a915daa79228590a11c2db4312ae/src/python/pants/util/contextutil.py#L30) in #5490. That method might be a bit messy (and you might be able to rewrite it much better), but it would be great to have path handling methods in a single place so they can be shared across different parts of pants and continually improved. If you could dump this logic in a method in contextutil.py (or maybe process_handler.py?) and call it here, that would be cool, but if you don't want to in this pr I can do that myself in a new pr.

Reply:
done.